rt/rs/extensions/providers/src/main/java/org/apache/cxf/jaxrs/provider/json/JSONProvider.java

```diff
@@ -239,8 +239,8 @@ public T readFrom(Class<T> type, Type genericType, Annotation[] anns, MediaType
             } else {
                 response = checkAdapter(response, type, anns, false);
             }
+            realStream.close();
```

Comment:
So this may not work: the stream is not created here and should not be closed here

Reply:
Can we do this within a try-with-resources?

Reply:
I think it is being done this way (higher in the stack), when the request / response body is being processed

Reply:
@cunningt could you please revert this one change and we are good to go, as per spec:

> The caller is responsible for ensuring that the input stream ends when the entity has been consumed. The implementation should not close the input stream.